api/apps/restful_apis/chunk_api.py

```
@@ -1037,10 +1048,36 @@ async def add_chunk(tenant_id, dataset_id, document_id):
+    # (#18174). ``thread_pool_exec`` already wraps the call in
+    # ``loop.run_in_executor`` with a fresh ``ThreadPoolExecutor``; the
+    # outer ``asyncio.wait_for`` enforces the time bound.
+    encode_inputs = [doc.name, req["content"] if not d["question_kwd"] else "\n".join(d["question_kwd"])]
+    try:
+        v, c = await asyncio.wait_for(
+            thread_pool_exec(embd_mdl.encode, encode_inputs),
+            timeout=ADD_CHUNK_OPERATION_TIMEOUT,
+        )
+    except asyncio.TimeoutError:
+        return get_error_data_result(message=f"Embedding model timed out after {ADD_CHUNK_OPERATION_TIMEOUT}s.")
+    except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)